File: c++-src/py_capi/tests/tpg_network_test.py
#! /usr/bin/env python3

"""TpgNetwork のテストプログラム

:file: tpg_network_test.py
:author: Yusuke Matsunaga (松永 裕介)
:copyright: Copyright (C) 2023 Yusuke Matsunaga, All rights reserved.
"""

import pytest
import os
from druid.types import TpgNetwork, FaultType
from make_filename import make_filename


def test_read_blif():
    filename = make_filename('s38584.blif')
    network = TpgNetwork.read_blif(filename, FaultType.StuckAt)

    assert network.node_num == 22447
    assert network.input_num == 12
    assert network.output_num == 278
    assert network.ppi_num == 1464
    assert network.ppo_num == 1730
    assert network.mffc_num == 4689
    assert network.ffr_num == 5676
    assert network.dff_num == 1452
            
def test_read_bench():
    filename = make_filename('b01.bench')
    network = TpgNetwork.read_bench(filename, FaultType.StuckAt)
    
    assert network.node_num == 54
    assert network.input_num == 2
    assert network.output_num == 2
    assert network.ppi_num == 7
    assert network.ppo_num == 7
    assert network.mffc_num == 21
    assert network.ffr_num == 24
    assert network.dff_num == 5
    

File: c++-src/types/test-programs/tpgnetworktest/TpgNetworkTest.cc

/// @file TpgNetworkTest.cc
/// @brief TpgNetworkTest の実装ファイル
/// @author Yusuke Matsunaga (松永 裕介)
///
/// Copyright (C) 2017 Yusuke Matsunaga
/// All rights reserved.

#include "types/TpgNetwork.h"
#include "types/TpgFaultList.h"
#include "types/FaultType.h"


BEGIN_NAMESPACE_DRUID

void
loop_test(
  SizeType loop_count,
  const std::string& filename
)
{
  for ( SizeType i1 = 0; i1 < loop_count; ++ i1 ) {
    for ( SizeType i2 = 0; i2 < loop_count; ++ i2 ) {
      std::cout << "            "
		<< "\r" << i1 << ": " << i2 << "\r";
      std::cout.flush();
      try {
	auto network = TpgNetwork::read_iscas89(filename, FaultType::StuckAt);
	auto fault_list = network.rep_fault_list();
      }
      catch ( std::invalid_argument ) {
	std::cerr << "Error in reading " << filename
		  << std::endl;
	break;
      }
    }
  }
  std::cout << std::endl;
}

END_NAMESPACE_DRUID

int
main(
  int argc,
  char** argv
)
{
  using namespace std;

  if ( argc != 2 ) {
    cerr << "USAGE: " << argv[0] << " filename" << endl;
    return 255;
  }

  DRUID_NAMESPACE::loop_test(100, argv[1]);

  return 0;
}
